packages/mobile/src/components/audio/Audio.tsx

```diff
@@ -333,34 +359,27 @@ export const Audio = () => {
   )
   const offlineTrackUri = useOfflineTrackUri(track)
 
-  if (!track || track.is_delete) return null
-
-  const gateways = trackOwner
-    ? audiusBackendInstance.getCreatorNodeIPFSGateways(
-        trackOwner.creator_node_endpoint
-      )
-    : []
+  useEffect(() => {
+    if (track && trackOwner) {
+      setStreamUris(getStreamUris(track, trackOwner))
+    }
+  }, [track, trackOwner])
```

Comment:
Could be cleaner as a memo imo:

```suggestion
  const streamUris = useMemo(() => {
    if (track && trackOwner) {
      return getStreamUris(track, trackOwner)
    }
    return null
  }, [track, trackOwner])
```

Reply:
I need this to only call the state setter once per track update, since we're using the state as a queue to track which ones we haven't tried yet

Reply:
Ah gotcha, makes sense

Reply:
Though @piazzatron just brought up that we could use the Discovery endpoint's redirect instead. That might be better than doing the fallback in client?

Reply:
Oh yeah that could be better, fewer requests and also avoids error states in the `Video` component